addons/base_vat/models/res_partner.py

```diff
@@ -112,30 +114,44 @@ def simple_vat_check(self, country_code, vat_number):
             return bool(self.env['res.country'].search([('code', '=ilike', country_code)]))
         return check_func(vat_number)
 
+    @api.depends('vat', 'country_id')
+    def _compute_vies_vat_to_check(self):
+        """ Retrieve the the VAT number, if one such exists, to be used when checking against the VIES system """
+        eu_country_codes = self.env.ref('base.europe').country_ids.mapped('code')
+        for partner in self:
+            if not partner.vat:
+                partner.vies_vat_to_check = ''
+                continue
+            country_code, vat_split = partner._split_vat(partner.vat)
+            if not country_code.isalpha() and partner.country_id:
+                country_code = partner.country_id.code
+                vat_split = partner.vat
+            partner.vies_vat_to_check = (
+                country_code.upper() in eu_country_codes or
+                country_code.lower() in _region_specific_vat_codes
+            ) and country_code + vat_split or ''
+
+    @api.depends_context('company')
+    @api.depends('vies_vat_to_check')
+    def _compute_perform_vies_check(self):
+        """ Determine whether a VIES check is to be performed on the current VAT number """
+        if self.env.context.get('company_id'):
+            company = self.env['res.company'].browse(self.env.context['company_id'])
+        else:
+            company = self.env.company
```

Comment:
Why wouldn't taking the company on the env be enough?
This is not consistent with `_get_fiscal_position`